src/conf_mode/dns_forwarding.py

```diff
@@ -58,6 +58,7 @@ def get_config(config=None):
     default_values = defaults(base)
     # T2665 due to how defaults under tag nodes work, we must clear these out before we merge
     del default_values['authoritative_domain']
+    del default_values['name_server']
```

Comment:
One thing which is not clear to me.
You use defaultValue port 53 in the XML then delete it there.
Is there any place where it is used?

Reply:
Good call! Yes, I discovered the hard way that `defaults` doesn't seem to work the way you'd expect - particularly for `tagNode`. There is an [open issue](https://vyos.dev/T2665) regarding this.

If you, see the [line right](https://github.com/vyos/vyos-1x/blob/8f1e8d720d63dffa2b5a3413c8827c9ad176f2ac/src/conf_mode/dns_forwarding.py#L59-L60) above that line, the same workaround has been applied for `authoritative_domain` as well.

In this case, with the following commands:
```
set service dns forwarding name-server 1.1.1.1
set service dns forwarding name-server 1.0.0.1 port 853
```
This is what we would expect in `dns` object:
```
  "name_server": {
    "1.1.1.1": { "port": "53" },
    "1.0.0.1": { "port": "853" }
  },
```
Instead, what we get in `dns` object this:
```
  "name_server": {
    "1.1.1.1": {},
    "1.0.0.1": { "port": "853" },
    "port": "53"
  },
```
Notice that `"port": "53"` is dangling outside incorrectly :)

So all that `defaultValue` is doing is providing a _hint_ to the user in CLI. But not actually doing anything useful. This isn't ideal, but I am open to advice.

Reply:
For example https://github.com/vyos/vyos-1x/blob/8f1e8d720d63dffa2b5a3413c8827c9ad176f2ac/src/conf_mode/system_sflow.py#L60

Reply:
So looks like we have to explicitly apply the defaults by looping through the the `tagNode`s. I'll update this PR shortly. Thank you for the example!

Reply:
Updated, thanks for help @sever-sever!

Reply:
You discovered correctly @indrajitr defaultValues + tageNodes is a bit clumsy